pkg/node/biz/client.go

```diff
@@ -25,19 +25,33 @@ func join(peer *signal.Peer, msg proto.JoinMsg) (interface{}, *nprotoo.Error) {
 		return nil, ridError
 	}
 
+	islb, found := getRPCForIslb()
+	if !found {
+		return nil, util.NewNpError(500, "Not found any node for islb.")
+	}
+	uid := peer.ID()
+
 	//already joined this room
 	if signal.HasPeer(rid, peer) {
-		return emptyMap, nil
+		log.Infof("biz.join peer.ID()=%s already joined, removing old peer", peer.ID())
+		islb.AsyncRequest(proto.IslbOnStreamRemove, util.Map("rid", rid, "uid", uid))
+
+		_, err := islb.SyncRequest(proto.IslbClientOnLeave, util.Map("rid", rid, "uid", uid))
+		if err != nil {
+			log.Errorf("IslbClientOnLeave failed %v", err.Error())
+		}
+
+		oldPeer := signal.GetPeer(rid, peer.ID())
```

Comment:
is it possible that due to a race condition, this will be null?

Reply:
I haven't had that case yet, but a check shouldn't hurt, I'll add that.